Solved state 2 issue and Added new features

irahhari opened the pull request:
State was not becoming 2 when it should because it was being modulo-ed by 2 when the number of states is 3 https://github.com/digital-design-snu/RNBIP/issues/2

irahhari committed:
Solved state 2 issue

State was not becoming 2 when it should because it was being modulo-ed by 2 when the number of states is 3

irahhari committed:
pretty print hex value

created a static method to pretty print hex value: stripped prefix 0x and always made sure the length of the hex value as string is 2

irahhari committed:
Color change for updated value in the register

color of the value in the register changes to red when the value is updated. it remains black if there is no change

irahhari commented:
Added features:
- all hex value will be printed in only 2 characters
- Color changes to red when the register value is updated

command-paul commented:
Hey @irahhari , ref #5 for the my fix for the functional bug you were seeing, and like the Conditional Call instruction now uses the (operand register) OR -> PC path (enabled by (LPC && IPC)). All instructions execute in a max of two cycles, => state ∈{0,1}  which when concatenated with the opcode is sufficient to address all control bit patterns for this microarchitecture.

Thanks for fixing the cosmetic issue, If you could revert the state bit modification, and add your name to the Readme in a new contributors section I'd be more than happy to merge this PR.

ref.https://github.com/digital-design-snu/RNBIP/blob/master/rnbip_emulator/emulator_backend.py#L362

irahhari committed:
Update E_PC.png

irahhari commented:
Hello @command-paul, I closed issue #5. I have updated my fork to solve issue #4. With respect to issue #2, here is an example:

![Single-Bus Processor](https://user-images.githubusercontent.com/76170080/218277318-e9714e0e-04ce-43c4-b378-cb4a739a89d4.gif)

For this particular example, I have listed down the expected value and the actual value which I took from the GUI.

| | Name of Cycle| Expected Value | Actual Value (from example) |
|-|-|-|-|
|First state | Opcode Fetch | 0x00 | 0x00 |
|Second state | Operand Read | 0x01 |0x01 |
|Third state | Execute | 0x02 | 0x00|

It is clearly seen that there is conflict in the state value, and I have raised the same in #2 and fixed it in [81da106](https://github.com/digital-design-snu/RNBIP/pull/3/commits/81da106aeb89f61965900809c4c9011322607fb7)
